doc/source/packaging/build-systems.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_build_system:
+
+############
+Build System
+############
+
+The build system is a fundamental tool when it comes to packaging Python
+libraries. It allows to generate distribution files which can be shared with
+users and developers. 
+
+
+The Build System
```

Comment:
@jorgepiloto I'm a bit  confused as to what seems an interchangeable use of build system versus build system. Is there a way that this can be consistent or clarified?

Reply:
Thanks for pointing this out @jorgepiloto. You are right, having "build system" in the header of the page and in here is confusing.

We could rename this section as "Artifacts", as it is a short title and these are the end goal of the build system.

Reply:
My comment has been edited. I wanted to point out that there were instances of "build system" and "build-system" and I was hoping they could be made consistent.